maven-plugin/src/main/java/org/openmrs/maven/plugins/AbstractTask.java

```diff
@@ -166,6 +176,16 @@ public AbstractTask(AbstractTask other) {
 	}
 
 	public void initTask() {
+		if (mavenEnvironment == null) {
+			mavenEnvironment = new MavenEnvironment();
+			mavenEnvironment.setMavenProject(mavenProject);
+			mavenEnvironment.setMavenSession(mavenSession);
+			mavenEnvironment.setSettings(settings);
+			mavenEnvironment.setArtifactMetadataSource(artifactMetadataSource);
+			mavenEnvironment.setArtifactFactory(artifactFactory);
+			mavenEnvironment.setPluginManager(pluginManager);
+			mavenEnvironment.setWizard(wizard);
+		}
```

Comment:
See the new `MavenEnvironment` class below - basically this just allows us to encapsulate everything provided by the Maven process during plugin execution in a single object that can be passed around and depended upon.